src/main/java/org/jenkinsci/plugins/stashNotifier/NotifierSelectors/DefaultHttpNotifierSelector.java

```diff
@@ -0,0 +1,37 @@
+package org.jenkinsci.plugins.stashNotifier.NotifierSelectors;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.jenkinsci.plugins.stashNotifier.Notifiers.DefaultApacheHttpNotifier;
+import org.jenkinsci.plugins.stashNotifier.Notifiers.ExtendedApacheHttpNotifier;
+import org.jenkinsci.plugins.stashNotifier.Notifiers.HttpNotifier;
+import org.jenkinsci.plugins.stashNotifier.SelectionContext;
+
+import java.util.*;
+
+public class DefaultHttpNotifierSelector implements HttpNotifierSelector {
+    private final List<HttpNotifier> httpNotifiers;
+
+    public DefaultHttpNotifierSelector(List<HttpNotifier> notifiers) {
```

Comment:
Are there more than two implementations?

I like the idea here a lot. I think it would be quite a bit simpler by having the constructor take the `Default` and `Extended` notifiers directly, rather than iterating a list for a known value.